chia/rpc/wallet_rpc_api.py

```diff
@@ -586,7 +586,7 @@ async def create_new_wallet(self, request: Dict) -> EndpointResult:
 
             elif request["mode"] == "existing":
                 async with self.service.wallet_state_manager.lock:
-                    cat_wallet = await CATWallet.create_wallet_for_cat(
+                    cat_wallet = await CATWallet.get_or_create_wallet_for_cat(
                         wallet_state_manager, main_wallet, request["asset_id"], name
                     )
                 self.service.wallet_state_manager.state_changed("wallet_created")
```

Comment:
this state changed message will fire now in cases where the wallet wasn't actually created, but perhaps that is harmless

Reply:
Moved the notification inside of the function just in case